File: awa/cookbook/client-workers/java/src/main/java/com/slalom/awa/configuration/TemporalConfiguration.java
package com.slalom.awa.configuration;

public class TemporalConfiguration {
    private final String clientTargetHost;
    private final String clientNamespace;
    private final String taskQueue;

    public TemporalConfiguration() {
        this("localhost:7233", "default", "awa_client_java");
    }

    public TemporalConfiguration(String clientTargetHost, String clientNamespace, String taskQueue) {
        this.clientTargetHost = clientTargetHost;
        this.clientNamespace = clientNamespace;
        this.taskQueue = taskQueue;
    }

    public String getClientTargetHost() {
        return clientTargetHost;
    }

    public String getClientNamespace() {
        return clientNamespace;
    }

    public String getTaskQueue() {
        return taskQueue;
    }
}

File: awa/cookbook/client-workers/dotnet/AgenticWorkflowApp.Console/Configuration/TemporalConfiguration.cs
namespace AgenticWorkflowApp.Console.Configuration;

/// <summary>
/// Configuration options for Temporal client and worker settings.
/// </summary>
public class TemporalConfiguration
{
    public const string SectionName = "Temporal";

    /// <summary>
    /// Temporal server target host (e.g., localhost:7233)
    /// </summary>
    public string ClientTargetHost { get; set; } = "localhost:7233";

    /// <summary>
    /// Temporal namespace to use
    /// </summary>
    public string ClientNamespace { get; set; } = "default";

    /// <summary>
    /// Task queue for this worker to listen on
    /// </summary>
    public string TaskQueue { get; set; } = "awa_client_dotnet";
}
